HubSpot.NET/Api/OAuth/HubSpotOAuthApi.cs

```diff
@@ -0,0 +1,154 @@
+﻿namespace HubSpot.NET.Api.OAuth
+{
+    using HubSpot.NET.Api.OAuth.Dto;
+    using HubSpot.NET.Core;
+    using HubSpot.NET.Core.Abstracts;
+    using HubSpot.NET.Core.Interfaces;
+    using Newtonsoft.Json;
+    using RestSharp;
+    using RestSharp.Serializers;
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+
+    public class HubSpotOAuthApi : ApiRoutable, IHubSpotOAuthApi
+    {
+        public string ClientId { get; private set; }
+        private string _clientSecret;
+        private IHubSpotClient _client;
+
+        public override string MidRoute => "oauth/v1/token";
+
+        private Dictionary<OAuthScopes, string> OAuthScopeNameConversions = new Dictionary<OAuthScopes, string>
+        {
+            { OAuthScopes.Automation , "automation" },
```

Comment:
I'm not sure what you mean by saying there will be two places to maintain the list. Could you please clarify?

Reply:
Ah, I see what you mean now. Well since `enum`s are restricted to having numeric values, this dictionary maps a strongly typed OAuthScope with its string value.

Reply:
I considered that as well for each one. We could condense it further for simplicity into a single location.

Reply:
What I've seen in the past is something like the below - and an extension method to get the stringified version (and vice versa). What do you think?

```csharp
public enum MyEnum {
     [StringifiedName("option-one")
     OptionOne,

     [StringifiedName("option-two")
     OptionTwo
}
```

Reply:
I would need to make a test project to validate that it works with Json.Net. I believe the problem I had was when serializing the values it wasn't turning out how I wanted.

Reply:
No worries either way I don't think - these values are pretty static I figure.